gtk/steam-imcontext-gtk.c

```diff
@@ -129,3 +125,34 @@ static void steam_im_context_focus_in(GtkIMContext *context)
         g_warning("Error opening Steam keyboard: %s", error->message);
     }
 }
+
+static gboolean steam_im_context_filter_keypress(GtkIMContext *context, GdkEventKey *event)
+{
+    SteamIMContext *im_context = STEAM_IM_CONTEXT(context);
+
+    if (event->type == GDK_KEY_RELEASE) {
+        if (event->keyval == GDK_KEY_BackSpace) {
```

Comment:
You don't need to handle backspace/delete/and arrow keys: those are managed by GTK itself when returning FALSE from `filter_keypress`.

The only thing that's needed is the emission of the "commit" signal with the Unicode glyph converted into a string.